schema/types.go

```diff
@@ -77,3 +78,12 @@ type PodInfo struct {
 type ContainerInfo struct {
 	InfoType
 }
+
+// Supported metric names, used as keys for all map[string]*store.TimeStore
+const cpuLimit = "cpu/limit"
```

Comment:
All your points are valid! One option is to have a map of resource type to
a timeseries of resource metrics (spec + stats). This is closer to the
kubernetes model.
As long as this is not what we expose via APIs, I am totally fine deferring
the refactoring to a later stage.

On Wed, Jun 24, 2015 at 3:35 PM, Alexandros Mavrogiannis <
notifications@github.com> wrote:

> In schema/types.go
> <https://github.com/GoogleCloudPlatform/heapster/pull/365#discussion_r33204976>
> :
>
> > @@ -77,3 +78,12 @@ type PodInfo struct {
> >  type ContainerInfo struct {
> >  	InfoType
> >  }
> > +
> > +// Supported metric names, used as keys for all map[string]*store.TimeStore
> > +const cpuLimit = "cpu/limit"
>
> That is the eventual goal, however there are a few issues to consider,
> such as:
>
>    - time resolution of limits vs time resolution of usage.
>    - mismatch of limit/usage timestamps.
>    - input resolution versus output resolution for historical data.
>    - instantaneous derived stats (e.g. current utilization percentage)
>    and the required conversions of limits/usage values.
>
> For those reasons, I believe it's better if we address the format of the
> resource struct together with the design of the derived stats.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/heapster/pull/365/files#r33204976>
> .
>